### PR TITLE
some update to Device Memory API

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

initial the documentation of `WorkerNavigator.deviceMemory`

specific the `spec-urls` meta data member of Device Memory API api summary page

add *Extensions to other interfaces* section to Device Memory API api summary page and so remove `Navigator.deviceMemory` from *See also* section

add Device Memory API to API group data and so use `{{APIRef("Device Memory API")}}` macros

mark experimental `false` of `navigator.deviceMemory` and `Device-Memory` header, see https://github.com/mdn/browser-compat-data/pull/21350

add `{{AvailableInWorkers}}` to Device Memory API summary page, since this API can both used in Window and Worker

remove the incorrect mark of `{{deprecated_inline}}` of `Device-Memory` header in http headers summary

some other style update

### Additional details

Working on https://developer.mozilla.org/en-US/docs/Web/API/Device_Memory_API

Reference to https://www.w3.org/TR/device-memory/

### Related issues and pull requests

Depends on: https://github.com/mdn/browser-compat-data/pull/21350

This is part of #22441

### CODE REVIEW

File: files/en-us/web/api/workernavigator/devicememory/index.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+## Value
+
+A floating point number; one of `0.25`, `0.5`, `1`, `2`, `4`, `8`.
+
+## Examples
+
+```js
+const memory = navigator.deviceMemory;
+console.log(`This device has at least ${memory}GiB of RAM.`);
```
Comment (teoli2003) [inline files/en-us/web/api/workernavigator/devicememory/index.md:28]: I think this example should be inside a worker.

**Review by teoli2003 — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/workernavigator/devicememory/index.md (hunk 1 above)
Comment (skyclouds2001) [reply files/en-us/web/api/workernavigator/devicememory/index.md:30]: add a note to suggest the code can be run in worker

File: files/en-us/web/api/device_memory_api/index.md [hunk 2]
```diff
@@ -17,7 +19,7 @@ There are two ways to determine the approximate amount of RAM a device has: use
 
 ### JavaScript API
 
-You may query the approximate amount of RAM a device has by retrieving {{DOMxRef("Navigator.deviceMemory")}}
+You may query the approximate amount of RAM a device has by retrieving {{DOMxRef("Navigator.deviceMemory")}} or {{DOMxRef("WorkerNavigator.deviceMemory")}}
```
Comment (teoli2003) [inline files/en-us/web/api/device_memory_api/index.md:22]:
```suggestion
You may query the approximate amount of RAM a device has by retrieving {{DOMxRef("Navigator.deviceMemory")}} or {{DOMxRef("WorkerNavigator.deviceMemory")}}.
```

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**